src/extension/sdk/dev_tools/manager.ts

```diff
@@ -102,7 +104,16 @@ export class DevToolsManager implements vs.Disposable {
 				return undefined;
 			}
 			this.capabilities.version = installedVersion;
-			if (silent) {
+			if (flutterDaemon != null) {
+				const result = await flutterDaemon.serveDevTools();
+				this.devtoolsUrl = new Promise<string>((resolve, reject) => {
+					if (result.host != null && result.port != null) {
```

Comment:
Can `host` and `port` be explicitly `null`, or would they be missing from the JSON? If missing, I these might need to check `undefined` instead (or both?).

I usually to rely on thruthiness for simple things like this to avoid needing to be explicit about the difference - eg `if (result.host && result.port)` (it doesn't work if falsey values like `0` are valid though).